test/e2e/safeTypeNames.ts

```diff
@@ -0,0 +1,47 @@
+export const input = {
+  "title": "AnyOf",
+  "type": "object",
+  "properties": {
+    "foo": {
+      'type': 'object',
+      anyOf: [
+        {$ref: "#/definitions/stra'nge#name"},
+        {$ref: "#/definitions/keepUPPERCASE"},
+        {$ref: "#/definitions/under_score"},
+        {$ref: "#/definitions/camelCase"},
+        {$ref: "#/definitions/kebab-case"}
```

Comment:
Can you also add test cases for:

- Names that contain or start with spaces
- Names that contain or start with numbers
- Names that contain or start with periods
- Names that contain or start with commas
- Names that contain or start with *, $, or !
- Names that start with uppercase characters
- Words that are all uppercase